src/main/java/htsjdk/samtools/util/zip/DeflaterFactory.java

```diff
@@ -43,36 +43,21 @@
  */
 public class DeflaterFactory {
 
-    private static Constructor<IntelDeflater> intelDeflaterConstructor;
+    private static final boolean isUsingIntelDeflater;
 
     static {
-        try {
-            if (Defaults.TRY_USE_INTEL_DEFLATER) {
-                final Class<IntelDeflater> clazz = (Class<IntelDeflater>) Class.forName("htsjdk.samtools.util.zip.IntelDeflater");
-                intelDeflaterConstructor = clazz.getConstructor(Integer.TYPE, Boolean.TYPE);
-            }
-        } catch (ClassNotFoundException e) {
-            intelDeflaterConstructor = null;
-        } catch (NoSuchMethodException e) {
-            intelDeflaterConstructor = null;
-        } catch (UnsatisfiedLinkError e) {
-            intelDeflaterConstructor = null;
-        }
+        isUsingIntelDeflater = Defaults.TRY_USE_INTEL_DEFLATER && (new IntelDeflater().load());
```

Comment:
Exactly - it would cause a linking error during class loading, which is not something that can be recovered from, and then the DeflatorFactory would fail to load, and things would go badly.